### PR TITLE
Viet login update

### PR DESCRIPTION
**Pull request by saigonfox**

### CODE REVIEW

File: web-dev/src/components/LoginPage/index.js [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
-        if (returnURL) {
-            return path[1];
+    const [navigateLink] = useSearchParams();
+    const returnURL = navigateLink.get('returnURL');
+    const emailRef = useRef();
+    const passwordRef = useRef();
+    const navigate = useNavigate();
+    const {signIn} = useAuth()
+    const handleSignin = async () => {
+        try {
```
Comment (CrimpCity) [inline web-dev/src/components/LoginPage/index.js:16]: Can you add comments to the if try statements? For example "if this succeeds then the user is logged in and is sent to...else...."

File: web-dev/src/components/RegistrationPage/index.js [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
 const RegistrationPage = () => {
+    const firstNameRef = useRef();
+    const lastNameRef = useRef();
+    const userNameRef = useRef();
+    const emailRef = useRef();
+    const passwordRef = useRef();
+    const navigate = useNavigate();
+    const {signUp} = useAuth();
+    const handleSignup = async () => {
+        try {
```
Comment (CrimpCity) [inline web-dev/src/components/RegistrationPage/index.js:17]: Please add comments to the try block in order to make it more clear what the logic is. Thanks!

File: web-dev/src/components/data/Users.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "firstName": "Reigen",
+    "lastName": "Arataka",
+    "username": "SmileyKid",
+    "profilePicture": "../images/default_avatars/smiley.png",
+    "email": "Greatest.psychic@SpritisAndSuch.com",
+    "accountType": "Critic"
+  },
+  {
+    "firstName": "Kevin",
+    "lastName": "Nagumo",
```
Comment (CrimpCity) [inline web-dev/src/components/data/Users.json:12]: Who's Nagumo?

**Review by CrimpCity — APPROVED** [review]

File: web-dev/src/contexts/auth-context.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ import {login as doLogin, logout as doLogout, user as loadUser} from "../service
 const AuthContext = React.createContext();
 
 export const AuthProvider = ({children}) => {
+  const [profile, setProfile] = useState()
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/contexts/auth-context.js:7]: Why are you adding this? It has no use at all.

File: web-dev/src/components/RegistrationPage/index.js [hunk 5]
```diff
@@ -36,67 +60,89 @@ const RegistrationPage = () => {
...
-                                                        <label htmlFor="login-user-input"
+                                                    <label htmlFor="signup-username"
+                                                           className="login-user-label-title text-white-50">
+                                                        Username
+                                                    </label>
+                                                    <input type="text"
+                                                           ref={userNameRef}
+                                                           className="login-user-input text-white"
+                                                           id="signup-username"
+                                                           placeholder="Username"
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/components/RegistrationPage/index.js:88]: Add "required" as part of the username input component

File: web-dev/src/components/RegistrationPage/index.js [hunk 6]
```diff
@@ -36,67 +60,89 @@ const RegistrationPage = () => {
...
+                                                        <label htmlFor="signup-user-input"
                                                                className="login-user-label-title text-white-50">
                                                             Email or phone number
                                                         </label>
                                                         <input type="text"
+                                                               ref={emailRef}
                                                                className="login-user-input text-white"
-                                                               id="login-user-input"
+                                                               id="signup-user-input"
                                                                placeholder="Email or phone number"
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/components/RegistrationPage/index.js:105]: Add "required" as part of the emailOrNumber input component

File: web-dev/src/components/RegistrationPage/index.js [hunk 7]
```diff
@@ -36,67 +60,89 @@ const RegistrationPage = () => {
...
                                                            className="login-user-label-title text-white-50">
                                                         Password
                                                     </label>
                                                     <input type="password"
+                                                           ref={passwordRef}
                                                            className="login-user-input text-white"
-                                                           id="login-user-input"
+                                                           id="signup-password"
                                                            placeholder="Password"
                                                     />
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/components/RegistrationPage/index.js:123]: Add "required" as part of the password input component

File: web-dev/src/components/LoginPage/index.js [hunk 8]
```diff
@@ -31,21 +53,24 @@ const LoginPage = () => {
                                                             Email or phone number
                                                         </label>
                                                         <input type="text"
+                                                               ref={emailRef}
                                                                className="login-user-input text-white"
                                                                id="login-user-input"
                                                                placeholder="Email or phone number"
                                                         />
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/components/LoginPage/index.js:60]: Add "required" as part of the emailOrNumber input component

File: web-dev/src/components/LoginPage/index.js [hunk 9]
```diff
@@ -31,21 +53,24 @@ const LoginPage = () => {
...
                                         <div className="login-user-frame">
                                             <div className="position-relative">
                                                 <div className="login-email-box">
-                                                    <label htmlFor="login-user-input"
+                                                    <label htmlFor="login-password"
                                                            className="login-user-label-title text-white-50">
                                                         Password
                                                     </label>
                                                     <input type="password"
+                                                           ref={passwordRef}
```
Comment (ChokchaiThanrukprasert) [inline web-dev/src/components/LoginPage/index.js:73]: Add "required" as part of the password input component

**Pull request merged**